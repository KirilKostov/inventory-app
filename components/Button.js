import React from "react";
import { View, StyleSheet, Pressable } from "react-native";
import { Text } from "./Text";
import Colors from "../constants/colors";

export const Button = (props) => {
  const { onPress, title = "" } = props;
  return (
    <Pressable
      style={({ pressed }) => [
        pressed && { opacity: 0.7 },
        styles.button,
        props.style,
      ]}
      onPress={onPress}
      disabled={props.disabled}
    >
      <Text style={{ ...styles.text, ...props.textStyle }}>{title}</Text>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  button: {
    alignItems: "center",
    justifyContent: "center",
    paddingVertical: 12,
    paddingHorizontal: 32,
    borderRadius: 4,
    backgroundColor: Colors.blue,
    shadowColor: "black",
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowRadius: 2,
    shadowOpacity: 0.4,
    elevation: 3,
  },
  text: {
    fontSize: 16,
    lineHeight: 21,
    letterSpacing: 0.25,
    color: "white",
  },
});
