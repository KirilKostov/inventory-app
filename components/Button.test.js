import React from "react";
import renderer from "react-test-renderer";

import { Button } from "./Button";

describe("<Button />", () => {
  it("renders correctly", () => {
    const testRenderer = renderer.create(<Button />).toJSON();
    expect(testRenderer).toMatchSnapshot();
  });
  it("displays the correct title", () => {
    const testRenderer = renderer.create(<Button title="Test" />);
    const testInstance = testRenderer.root;
    expect(testInstance.props.title).toBe("Test");
  });
});
