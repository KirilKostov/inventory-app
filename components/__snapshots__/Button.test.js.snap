// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      false,
      Object {
        "alignItems": "center",
        "backgroundColor": "#0067ff",
        "borderRadius": 4,
        "elevation": 3,
        "justifyContent": "center",
        "paddingHorizontal": 32,
        "paddingVertical": 12,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.4,
        "shadowRadius": 2,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "gibson-regular",
        "fontSize": 16,
        "letterSpacing": 0.25,
        "lineHeight": 21,
      }
    }
  >
    
  </Text>
</View>
`;
